File: PhysicalDeviceTester/app/src/main/java/prateek_gupta/physical_device_tester/MainActivity.java
package prateek_gupta.physical_device_tester;

import android.app.ActivityOptions;
import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.transition.Fade;
import android.transition.Transition;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;

import prateek_gupta.physical_device_tester.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    ActivityMainBinding binding;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding=ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        LinearLayoutManager layoutManager=new LinearLayoutManager(this);
        binding.recyclerView.setItemViewCacheSize(-1);
        binding.recyclerView.setLayoutManager(layoutManager);
        binding.recyclerView.setAdapter(new Adapter(layoutManager));

        new Handler().postDelayed(()->{
            Intent intent=new Intent(MainActivity.this, MainActivity2.class);
            MainActivity.this.startActivityForResult(intent, 0,
                    ActivityOptions.makeCustomAnimation(MainActivity.this,
                            R.anim.stay,R.anim.stay).toBundle());
        },1000);
    }
}

File: PhysicalDeviceTester/app/src/main/java/prateek_gupta/physical_device_tester/MainActivity.kt
package prateek_gupta.physical_device_tester

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import prateek_gupta.physical_device_tester.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        binding.instagram.setOnClickListener { launchApp("http://instagram.com/_u/gupta__prateek") }
        binding.linkedIn.setOnClickListener { launchApp("https://www.linkedin.com/in/prateek-gupta-5a8122196/") }
        binding.github.setOnClickListener { launchApp("https://github.com/prateekgupta25apr") }
        binding.hackerrank.setOnClickListener { launchApp("https://www.hackerrank.com/prateekgupta2504?hr_r=1") }
        binding.stackoveflow.setOnClickListener { launchApp("https://stackoverflow.com/users/13329963/prateek-gupta") }
        binding.twitter.setOnClickListener { launchApp("https://twitter.com/GuPtA__PraTeeK") }
        binding.facebook.setOnClickListener { launchApp("fb://facewebmodal/f?href=https://www.facebook.com/prateek.gupta.12764/") }
    }

    private fun launchApp(url: String?) {
        val uri = Uri.parse(url)
        val intent = Intent(Intent.ACTION_VIEW)
        intent.data = uri
        startActivity(intent)
    }
}